babel.config.js

```diff
@@ -0,0 +1,19 @@
+const fs = require("fs");
+
+/**
+ * @summary Babel 7 doesn't transpile files in sub-directories that have a package.json when Babel is configured
+ * through package.json or .babelrc. This causes Jest test failures in custom plugins that have a package.json. It
+ * isn't an issue with non-test files because those are imported through the main Reaction app (see /server/plugins.js
+ * and client/plugins.js). Babel does transpile these files when it is configured through the new babel.config.js.
+ * Meteor currently only loads Babel config through .babelrc or package.json. So, in order to support Babel transpiling
+ * of Jest tests, we load the babel config defined in package.json and export it here.
+ * See this Github comment: https://github.com/facebook/jest/issues/6053#issuecomment-383632515
+ */
+module.exports = function (api) {
+  api.cache(false);
+
+  const file = fs.readFileSync("./package.json");
```

Comment:
I'd like to add a note here mentioning that this is required because as you mentioned
>  Meteor only reads the babel config from .babelrc or package.json. So with just babel.config.js, the Meteor app fails but the Jest tests pass. With just package.json, the Jest tests in custom plugins fail but the app runs.

So that we're aware of why we did this and if Meteor changes or our dependency on Meteor changes we can change this in the future.

Reply:
@spencern Comment added